File: app/src/main/java/ru/santaev/clipboardtranslator/viewmodel/ChooseLanguageViewModel.java
package ru.santaev.clipboardtranslator.viewmodel;


import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.schedulers.Schedulers;
import ru.santaev.clipboardtranslator.db.entity.Language;
import ru.santaev.clipboardtranslator.model.IDataModel;

public class ChooseLanguageViewModel extends ViewModel {

    private MutableLiveData<List<Language>> languages = new MutableLiveData<>();
    private IDataModel dataModel;
    private CompositeDisposable compositeDisposable = new CompositeDisposable();

    public ChooseLanguageViewModel(IDataModel dataModel) {
        this.dataModel = dataModel;
        this.languages.setValue(new ArrayList<>());

        compositeDisposable.add(dataModel.getLanguages()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(this::onLanguagesChanged));
    }

    private void onLanguagesChanged(List<Language> languages) {
        this.languages.postValue(languages);
    }

    public LiveData<List<Language>> getLanguages() {
        return languages;
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        compositeDisposable.dispose();
    }
}

File: app/src/main/java/ru/santaev/clipboardtranslator/viewmodel/ChooseLanguageViewModel.kt
package ru.santaev.clipboardtranslator.viewmodel


import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.santaev.domain.dto.LanguageDto
import com.example.santaev.domain.factory.UseCaseFactory
import io.reactivex.disposables.CompositeDisposable
import ru.santaev.clipboardtranslator.util.RxHelper
import java.util.*

class ChooseLanguageViewModel() : ViewModel() {

    private val compositeDisposable = CompositeDisposable()
    val languages = MutableLiveData<List<LanguageDto>>()

    init {
        this.languages.value = ArrayList()

        val disposable = UseCaseFactory
                .instance
                .getGetLanguagesUseCase()
                .execute()
                .compose(RxHelper.getFlowableTransformer())
                .subscribe(
                        { onLanguagesChanged(it) },
                        { onLanguagesChanged(listOf()) }
                )
        compositeDisposable.add(disposable)
    }

    private fun onLanguagesChanged(languages: List<LanguageDto>) {
        this.languages.postValue(languages)
    }

    override fun onCleared() {
        super.onCleared()
        compositeDisposable.dispose()
    }
}
